include/xlnt/xlnt_config.hpp

```diff
@@ -42,3 +42,17 @@
 #define XLNT_API
 #endif
 #endif
+
+#ifdef XLNT_EXPOSE_INTERNAL_API
+#define XLNT_API_INTERNAL XLNT_API
+#else
+#define XLNT_API_INTERNAL
+#endif
+
+#ifdef _MSC_VER
+#define XLNT_DEPRECATED __declspec(deprecated)
```

Comment:
If C++14 is available, shouldn't we default to [`[[deprecated]]` from C++14](https://en.cppreference.com/w/cpp/language/attributes/deprecated) instead of using compiler-specific code? I would keep all the code as it is, and if `XLNT_HAS_CPP_VERSION(XLNT_CPP_14)`, then `#define XLNT_DEPRECATED [[deprecated]]`, otherwise fall back to your code in C++11 mode.

Reply:
I changed it in my latest commit. I have reorganized some files to be able to use `XLNT_HAS_CPP_VERSION` inside `xlnt_config.hpp` (which is in the public include folder), without exposing too much implementation details.